src/icon-loader.c

```diff
@@ -101,37 +143,35 @@ icon_loader_lookup(struct server *server, const char *app_id, int size, int scal
 		icon_name = app_id;
```

Comment:
This creates an interesting case where an application can set its app_id to an absolute path and get it loaded as its icon. An absolute path must only come from `sfdo_desktop_entry_get_icon()`.

Reply:
Yes, that's a weird one. Have tweaked logic to avoid that.